src/AutoRecordings.cpp

```diff
@@ -248,8 +253,104 @@ PVR_ERROR AutoRecordings::SendAutorecAdd(const PVR_TIMER &timer)
 
 PVR_ERROR AutoRecordings::SendAutorecUpdate(const PVR_TIMER &timer)
 {
-  /* Note: there is no "updateAutorec" htsp method, thus delete + add. */
+  if (m_conn.GetProtocol() >= 25)
```

Comment:
Since the implementation is so differing between these different HTSP versions, maybe it would be clearer to delegate the actual logic to separate functions, i.e. something like this (pseudo):

```
AutoRecordings::SendAutoRecUpdate(timer) {
   if (htsp < 25) {
      SendAutoRecUpdateLegacy();
   } else {
      SendAutorecUpdateActual();
   }
}
```

Naming scheme is obviously up for debate.